distributed/comm/tests/test_comms.py

```diff
@@ -1313,30 +1310,18 @@ async def test_inproc_adresses():
     await check_addresses(a, b)
 
 
-def test_register_backend_entrypoint():
-    # Code adapted from pandas backend entry point testing
-    # https://github.com/pandas-dev/pandas/blob/2470690b9f0826a8feb426927694fa3500c3e8d2/pandas/tests/plotting/test_backend.py#L50-L76
```

Comment:
pandas `test_register_entrypoint` now looks like this: https://github.com/pandas-dev/pandas/blob/48d515958d5805f0e62e34b7424097e5575089a8/pandas/tests/plotting/test_backend.py#L47-L62

I used:

```
    monkeypatch.syspath_prepend(tmp_path)
    monkeypatch.setitem(sys.modules, "pandas_dummy_backend", dummy_backend)
```

in the pandas version of this PR  because they already do a lot of mutating of `sys.modules` so figured a `sys.path` mutation wasn't so bad. Here I use a `multiprocessing.Pool(1)` to isolate `sys.modules` and `sys.path` changes